### ISSUE TITLE
LCO observation form fails with 500 error

### ISSUE THREAD

**Issue by rachel3834** [LCOGT/mop#114]
For target Gaia21ccu, clicking on the LCO button on the Target Detail page, Observe tab causes a 500 error.  

Output from the stern logs shows the following:

mop-59cb5bfbc7-vdr2s mop Internal Server Error: /observations/LCO/create/
mop-59cb5bfbc7-vdr2s mop Traceback (most recent call last):
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
mop-59cb5bfbc7-vdr2s mop     response = get_response(request)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 222, in _get_response
mop-59cb5bfbc7-vdr2s mop     response = self.process_exception_by_middleware(e, request)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 362, in process_exception_by_middleware
mop-59cb5bfbc7-vdr2s mop     response = middleware_method(request, exception)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/tom_common/middleware.py", line 29, in process_exception
mop-59cb5bfbc7-vdr2s mop     raise exception
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 220, in _get_response
mop-59cb5bfbc7-vdr2s mop     response = response.render()
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/response.py", line 114, in render
mop-59cb5bfbc7-vdr2s mop     self.content = self.rendered_content
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/response.py", line 92, in rendered_content
mop-59cb5bfbc7-vdr2s mop     return template.render(context, self._request)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/backends/django.py", line 61, in render
mop-59cb5bfbc7-vdr2s mop     return self.template.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 175, in render
mop-59cb5bfbc7-vdr2s nginx 10.100.26.86 - - [13/Dec/2023:21:56:44 +0000] "GET /observations/LCO/create/?target_id=5215 HTTP/1.1" 500 145 "https://mop.lco.global/targets/5215/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36" "89.187.187.87"
mop-59cb5bfbc7-vdr2s mop     return self._render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 167, in _render
mop-59cb5bfbc7-vdr2s mop     return self.nodelist.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in render
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in <listcomp>
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 966, in render_annotated
mop-59cb5bfbc7-vdr2s mop     return self.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/loader_tags.py", line 157, in render
mop-59cb5bfbc7-vdr2s mop     return compiled_parent._render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 167, in _render
mop-59cb5bfbc7-vdr2s mop     return self.nodelist.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in render
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in <listcomp>
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 966, in render_annotated
mop-59cb5bfbc7-vdr2s mop     return self.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/loader_tags.py", line 63, in render
mop-59cb5bfbc7-vdr2s mop     result = block.nodelist.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in render
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in <listcomp>
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 966, in render_annotated
mop-59cb5bfbc7-vdr2s mop     return self.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/library.py", line 278, in render
mop-59cb5bfbc7-vdr2s mop     return t.render(new_context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 177, in render
mop-59cb5bfbc7-vdr2s mop     return self._render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 167, in _render
mop-59cb5bfbc7-vdr2s mop     return self.nodelist.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in render
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 1005, in <listcomp>
mop-59cb5bfbc7-vdr2s mop     return SafeString("".join([node.render_annotated(context) for node in self]))
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/base.py", line 966, in render_annotated
mop-59cb5bfbc7-vdr2s mop     return self.render(context)
mop-59cb5bfbc7-vdr2s mop   File "/usr/local/lib/python3.10/site-packages/django/template/library.py", line 258, in render
mop-59cb5bfbc7-vdr2s mop     _dict = self.func(*resolved_args, **resolved_kwargs)
mop-59cb5bfbc7-vdr2s mop   File "/mop/mop/templatetags/mop_extras.py", line 450, in classification_form
mop-59cb5bfbc7-vdr2s mop     if any(request.GET.get(x) for x in ['classification', 'text_class', 'category', 'text_category']):
mop-59cb5bfbc7-vdr2s mop   File "/mop/mop/templatetags/mop_extras.py", line 450, in <genexpr>
mop-59cb5bfbc7-vdr2s mop     if any(request.GET.get(x) for x in ['classification', 'text_class', 'category', 'text_category']):
mop-59cb5bfbc7-vdr2s mop AttributeError: 'str' object has no attribute 'GET'

**Comment by rachel3834** [comment]
I traced this issue to the target_data template partial that MOP has customized for displaying the parameters of a given Target. I thought that this partial was called only from the target detail page, so I only updated that view with the new target classifications form.  Turns out the target_data partial is also used in the LCO observe form...with a different number of arguments, hence the error.  

To make this more robust, I've separated the template partials used for the target detail page versus that for the observe form, and now the classification form is only included on the target detail page which makes more sense I think.  I've updated the corresponding templatetag to more robustly handle the different URL request routes by which it can be called.
